lhotse/dataset/sampling/data_source.py

```diff
@@ -18,6 +18,27 @@ def __init__(self, items: CutSet):
         self._shuffled_items = self._orig_items
         self._iter = None
         self._reusable = deque()
+        # Add duration tracking for non-lazy CutSets
+        if not self.is_lazy:
+            self._total_duration = sum(c.duration for c in self._orig_items)
+            self._total_cuts = len(self._orig_items)
+        else:
+            self._total_duration = None
+            self._total_cuts = None
+        self._remaining_duration = self._total_duration
+        self.remaining_cuts = self._total_cuts
+
+    @property
+    def is_lazy(self) -> bool:
+        return self._orig_items.is_lazy
+
+    @property
+    def remaining_duration(self) -> Optional[float]:
```

Comment:
Is there a reason why this is a property and not a function?  E.g. does it indicate that it's expected to be fast to compute?

Reply:
Yeah, that's the reason.